manifests/config.pp

```diff
@@ -85,6 +86,15 @@
     mode    => '0644',
   }
 
+  if $datalogstore and ($datalogstore != $datastore) {
```

Comment:
The comparison logic can be removed (here and in config below). if the default for $datalogstore is undef.